core/src/Cuda/Kokkos_Cuda_abort.hpp

```diff
@@ -74,8 +80,12 @@ namespace Impl {
   // This loop is never executed. It's intended to suppress warnings that the
   // function returns, even though it does not. This is necessary because
   // __assertfail is not marked as [[noreturn]], even though it does not return.
+  //  Disable with KOKKOS_ENABLE_DEBUG_BOUNDS_CHECK to workaround failures
+  //  in random number generator unit tests with pre-volta architectures
+#if !defined(KOKKOS_ENABLE_DEBUG_BOUNDS_CHECK)
```

Comment:
Guarding out the `while` loop here was necessary to resolve the unit test failures with the rand num gen; the other changes helped suppress warnings